src/tools/kata-ctl/src/check.rs

```diff
@@ -197,6 +208,110 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &KernelModule) -> Result<()> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+
+    // Partial check w/ modinfo
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module.name.clone())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let status_modinfo = output_modinfo.status;
+
+    // Partial check w/ modprobe
+    let output_modprobe = Command::new(MODPROBE_PATH)
+        .arg(MODPROBE_PARAMETERS_DRY_RUN)
+        .arg(MODPROBE_PARAMETERS_FIRST_TIME)
+        .arg(module.name.clone())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let status_modprobe = output_modprobe.status;
+
+    // Complete check
+    if status_modinfo.success() && !status_modprobe.success() {
+        return Ok(());
+    }
+
+    match status_modprobe.code() {
+        Some(code) => {
+            // Module not found
+            if code != 0 {
+                let std_err = String::from_utf8_lossy(&output_modinfo.stderr);
+                let msg = format!("{} {}", ERR_LSMOD, std_err);
+                return Err(anyhow!(msg));
+            }
+
+            // module not loaded
+            let msg = format!("{} {}", ERR_MODPROBE, module.name);
+            Err(anyhow!(msg))
+        }
+        None => Err(anyhow!("Process terminated by signal")),
+    }
+}
+
+pub fn check_kernel_module_parameters(
+    module: &KernelModule,
+) -> std::result::Result<std::result::Result<std::string::String, anyhow::Error>, ()> {
+    const MODULES_PATH: &str = "/sys/module";
+    const CAT_PATH: &str = "/usr/bin/cat";
+
+    let list_results = module.params
+            .iter()
+            .map(|(parameter, value_expected)| -> Result<String>{
+
+                let module_path = format!(
+                    "{}/{}/parameters/{}",
+                    MODULES_PATH, module.name, parameter
+                );
+
+                // 'cat' the module parameter specified
+                let output = Command::new(CAT_PATH)
+                    .arg(module_path)
+                    .stdout(Stdio::piped())
+                    .output()?;
+
+                let status = output.status;
+
+                if !status.success() {
+                    match status.code() {
+                        Some(_code) => {
+                            let msg = format!(
+                                "The \"{}\" parameter seems to be missing in the \"{}\" kernel module.",
```

Comment:
Nit: If you use `{:?}` rather than `\"{}\"`, the value will be quoted automatically.

Reply:
@jodh-intel fixed, thanks!